src/storage/store/engine_tests.rs

```diff
@@ -0,0 +1,249 @@
+
+        let db = db::RocksDB::new(db_path.to_str().unwrap());
+        db.open().unwrap();
+
+        let shard_store = ShardStore::new(db);
+        ShardEngine::new(1, shard_store)
+    }
+
+    fn enable_logging() {
+        let env_filter =
+            EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("warn"));
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(env_filter)
+            .try_init();
+    }
+
+    fn from_hex(s: &str) -> Vec<u8> {
+        hex::decode(s).unwrap()
+    }
+
+    fn to_hex(b: &[u8]) -> String {
+        hex::encode(b)
+    }
+
+    fn state_change_to_shard_chunk(
+        shard_idx: u32,
+        block_number: u64,
+        change: ShardStateChange,
+    ) -> ShardChunk {
+        let mut chunk = default_shard_chunk();
+
+        chunk.header.as_mut().unwrap().shard_root = change.new_state_root;
+        chunk.header.as_mut().unwrap().height = Some(Height {
+            shard_index: shard_idx,
+            block_number: block_number,
+        });
+
+        //TODO: don't assume 1 transaction
+        chunk.transactions[0]
+            .user_messages
+            .extend(change.transactions[0].user_messages.iter().cloned());
+
+        chunk
+    }
+
+    fn default_shard_chunk() -> ShardChunk {
+        ShardChunk {
+            header: Some(ShardHeader::default()),
+            // TODO: eventually we won't hardcode one transaction here
+            transactions: vec![Transaction {
+                user_messages: vec![],
+                system_messages: vec![],
+                fid: 1234,
+                account_root: vec![5, 5, 6, 6], //TODO,
+            }],
+            hash: vec![],
+            votes: None,
+        }
+    }
+
+    fn default_message(text: &str) -> Message {
+        let private_key = SigningKey::from_bytes(
+            &SecretKey::from_hex(
+                "1000000000000000000000000000000000000000000000000000000000000000",
+            )
+            .unwrap(),
+        );
+
+        cli::compose_message(private_key, 1234, text, Some(0))
+    }
+
+    fn entities() -> (Message, Message) {
+        let msg1 = default_message("msg1");
+        let msg2 = default_message("msg2");
+
+        assert_eq!(
+            "eb1850b43b2dd25935222c9137f5fa71b02b9689",
+            to_hex(&msg1.hash),
+        );
+
+        assert_eq!(
+            "ee0fcb6344d22ea2af4f97859108eb5a3c6650fd",
+            to_hex(&msg2.hash),
+        );
+
+        (msg1, msg2)
+    }
+
+    #[test]
+    fn test_engine_basic_propose() {
+        let mut engine = new_engine();
+        let state_change = engine.propose_state_change(1);
+
+        assert_eq!(1, state_change.shard_id);
+        assert_eq!(state_change.transactions.len(), 1);
+        assert_eq!(0, state_change.transactions[0].user_messages.len());
+        assert_eq!(
+            "237b11d0dd9e78994ef2f141c7f170d48bb51d34",
+            to_hex(&state_change.new_state_root)
```

Comment:
Hardcoding hashes is going to be fragile, especially since everything is in flux. I think we should just check that the root hash in the merkle trie matches the hash in the state change struct and not care about the actual values.

Reply:
Yes, agree with all above.

But I actually want this to be quite fragile to start, if the values change for any reason it will be interesting to see why they're changing and will require manual investigation and updates.

I'm happy to leave a comment to future editors of the test to feel free to do what you suggest above becomes too tedious and we have confidence with the code.